gameplay.c

```diff
@@ -130,14 +165,26 @@ void findTargets(int index = 0, string db = "") {
 	float dist = xsPow(yGetVarByIndex("allUnits", "range", index) * 6 + 1, 2);
 	int p = 3 - yGetVarByIndex("allUnits", "player", index);
 	trVectorQuestVarSet("pos", kbGetBlockPosition(""+1*yGetUnitAtIndex("allUnits", index), true));
+	bool anyTargets = false;
 	for(x=yGetDatabaseCount("allUnits"); >0) {
 		yDatabaseNext("allUnits");
 		if (yGetVar("allUnits", "player") == p) {
 			if (zDistanceToVectorSquared("allUnits", "pos") < dist) {
+				anyTargets = true;
 				yAddToDatabase(db, "allUnits");
+				// Bot
+				if(Bot){
+					trVectorSetUnitPos("botRightClick", "allUnits");
+				}
 			}
 		}
 	}
+	if(anyTargets){
+		// Bot attacks
+		if(Bot && trQuestVarGet("p2jobDoneHand") == 1){
+			BotClickRight();
+		}
+	}
 }
 
```

Comment:
Everything here can be achieved without modifying findTargets().

1. The bot calls the findTargets() function
2. To check if there are anyTargets, just check the size of the database created by findTargets. yGetDatabaseCount() > 0
3. Then the bot picks a target by using a database search and its own selection criteria.